### PR TITLE
Fix OpenACC backend codegen for constructor and destructor

### PR DESCRIPTION
**Pull request by iomaganaris**

### CODE REVIEW

**Review by kotsaloscv — APPROVED** [review]
tested & works

File: src/codegen/codegen_c_visitor.cpp [hunk 1]
```diff
@@ -3330,6 +3330,7 @@ void CodegenCVisitor::print_nrn_constructor() {
         const auto& block = info.constructor_node->get_statement_block();
         print_statement_block(*block.get(), false, false);
     }
+    print_kernel_data_present_annotation_block_end();
```
Comment (olupton) [inline src/codegen/codegen_c_visitor.cpp:3333]: GitHub doesn't let me comment where I want to comment...but: I am pretty sure the real problem is higher up this file, inside `print_global_function_common_code`, where we have:
```c++
if (type != BlockType::Destructor || type != BlockType::Constructor) {
  // We do not (currently) support DESTRUCTOR and CONSTRUCTOR blocks
  // running anything on the GPU.
  print_kernel_data_present_annotation_block_begin();
}
```
so the intention is clearly to not call `print_kernel_data_present_annotation_block_begin()` for constructors and destructors...but the condition is wrong. I think the correct fix is to replace `||` with `&&` there, not to call `print_kernel_data_present_annotation_block_end()` here.

File: src/codegen/codegen_c_visitor.cpp [hunk 2]
```diff
@@ -3341,6 +3342,7 @@ void CodegenCVisitor::print_nrn_destructor() {
         const auto& block = info.destructor_node->get_statement_block();
         print_statement_block(*block.get(), false, false);
     }
+    print_kernel_data_present_annotation_block_end();
```
Comment (olupton) [inline src/codegen/codegen_c_visitor.cpp:3345]: See above, I don't think we want this change.

**Review by olupton — CHANGES_REQUESTED** [review]

File: src/codegen/codegen_c_visitor.cpp (hunk 1 above)
Comment (iomaganaris) [reply src/codegen/codegen_c_visitor.cpp]: I think you're right Olli, yes :+1: I applied this change instead

**Review by olupton — APPROVED** [review]
LGTM, but the CI doesn't cover this at the moment so someone should check it locally.

**Review by kotsaloscv — APPROVED** [review]
I re-tested the amendement locally/BB5 (Olli's suggestion) and everything works fine.

**Review by alkino — APPROVED** [review]
Sorry for that :$

**Pull request merged**